### ISSUE TITLE
Improvements based on 318 release

### ISSUE THREAD

**Issue by PavelPavlov46** [gardenlinux/gardenlinux#242]
**What would you like to be added**: Based on the issues faced during 318 release we should consider following improvements:

**General:**
- First and most important, Backlog, feature, release work management needs significant improvement with highest priority.
- Provide in the final image a consistent user (e.g. admin/gardener/root). FMI Add common static OS user 
https://github.com/gardenlinux/gardenlinux/issues/50
- Builds are not consistent (e.g. old builds may fail due to  HTTP-503s being returned from the debian snapshot repository?
- Improve pipeline 
<img width="1136" alt="Screenshot 2021-04-22 at 15 21 49" src="https://user-images.githubusercontent.com/40396270/115713246-76d7e800-a37e-11eb-848a-f4dcaa58910b.png">

**Azure specific**
- waagent seems to looking for ifup and ifdown commands. Can/should we install them?
- waagent depends on older python versions and had to be forked and patched in https://github.com/gardenlinux/waagent. How do we push our fix back to Microsoft and avoid using own version? See https://github.com/gardenlinux/gardenlinux/issues/242#issuecomment-825528417.
- https://github.com/gardenlinux/gardenlinux/blob/rel-318/features/azure/exec.config
- Create custom user data test on Garden Linux or integration test on gardener side?

**Why is this needed**: In order to provide smoother releases in future.

**Comment by PavelPavlov46** [comment]
Regarding the azure linux agent (waagent) python problem, according to Azure team:

---

Issue: When trying to create a VM from the latest Garden Linux VM image version 318.4.0 it goes to OSPTO state i.e OSProvisioning State. This is a bug with the Azure Linux Agent, calling an attribute "toString" that was removed since Python 3.2.

https://github.com/Azure/WALinuxAgent/issues/2080 - PG is already rolling out a fix for it that comes in version 2.2.54, but it's a pre-release right now so not made available for install. (see: https://github.com/Azure/WALinuxAgent/releases/tag/pre-v2.2.54.1). Once this is fully released(no ETA provided), you would need to re-create this image with the new agent so it doesn't fail the provisioning. If you can't wait, you would need to rebuild and use a version of Python older than 3.2 at the moment.

More details with regards to this issue:
WALA provisioning failed: AttributeError: 'array.array' object has no attribute 'tostring' · Issue #2080 · Azure/WALinuxAgent
Description of problem:
WALA provisioning failed because of "AttributeError: 'array.array' object has no attribute 'tostring'"
The root cause is in python3.9 array.array doesn't have tostring(), should use tobytes() instead.
(In azurelinuxagent/common/osutil/default.py _get_all_interfaces function)
Version-Release number of selected components (if applicable):
WALinuxAgent-2.2.48.1-2.el9.noarch

---

**Comment by vpnachev** [comment]
> that was removed since Python 3.2.
> ...
> would need to rebuild and use a version of Python older than 3.2 at the moment.

There are some wrong statements from the Azure team, this should be read as 
```
that was deprecated since Python 3.2 and removed in 3.9
...
would need to rebuild and use a version of Python 3.8 or older at the moment.
```

**Comment by marwinski** [comment]
There are two many wishes in this ticket. In future, can we have separate issues for them. Anyway to 

> Provide in the final image a consistent user (e.g. admin/gardener/root). FMI Add common static OS user

The problem is that we inherited the username `admin` from Debian but as we found out later this is forbidden for Azure and GCP. I am not sure we can easily change it to something more useful like `garden`.

**Comment by marwinski** [comment]
I will close this ticket in favour of other tickets:

> First and most important, Backlog, feature, release work management needs significant improvement with highest priority.

I am currently doing this.

> Provide in the final image a consistent user (e.g. admin/gardener/root). FMI Add common static OS user

See comment above.

> Add common static OS user  #50

Won't fix. See discussion in the linked ticket.

> Builds are not consistent (e.g. old builds may fail due to HTTP-503s being returned from the debian snapshot repository?

Currently being addressed in #321  (or at least meant to be addressed. 

> Improve pipeline

This is addressed by Christian and his team.

> waagent

from what I can see we still don't have the fix in Debian. I have opened #322 to follow up.
